aten/src/ATen/native/cuda/DepthwiseConv3d.cu

```diff
@@ -0,0 +1,624 @@
+  }
+}
+
+}
+
+#define NODEF_OR_EQUAL(x, y) ((y) < 0 || (x) == (y))
+#define NODEF_OR_EQUAL_3(x, y1, y2, y3) \
+  (NODEF_OR_EQUAL(x[0], y1) && \
+   NODEF_OR_EQUAL(x[1], y2) && \
+   NODEF_OR_EQUAL(x[1], y3))
+
+#define DWCONV3D_FORWARD_DISPATCH_SPECIALIZATION(kt, kh, kw, dilt, dilh, dilw) \
+  if (NODEF_OR_EQUAL_3(kernel_size, (kt), (kh), (kw)) &&                    \
+      NODEF_OR_EQUAL_3(dilation, (dilt), (dilh), (dilw))) {                 \
+    using accscalar_t = acc_type<scalar_t, true>;                           \
+    conv_depthwise3d_cuda_kernel                                            \
+    <scalar_t, accscalar_t, (kt), (kh), (kw), (dilt), (dilh), (dilw)>       \
+      <<<grid, block, (smem), at::cuda::getCurrentCUDAStream()>>>(          \
+        input_.packed_accessor32<scalar_t, 5>(),                            \
+        output_.packed_accessor32<scalar_t, 5>(),                           \
+        weight_.packed_accessor32<scalar_t, 5>(),                           \
+        bias_ptr,                                                           \
+        stride[0], stride[1], stride[2],                                    \
+        padding[0], padding[1], padding[2],                                 \
+        dilation[0], dilation[1], dilation[2]);                             \
+  } else
+
+#define DWCONV3D_FORWARD_DISPATCH_OTHERS \
+  {                                      \
+    using accscalar_t = acc_type<scalar_t, true>;                           \
+    conv_depthwise3d_cuda_kernel                                            \
+    <scalar_t,accscalar_t, -1, -1, -1, -1, -1, -1>                          \
+      <<<grid, block, (smem), at::cuda::getCurrentCUDAStream()>>>(          \
+        input_.packed_accessor32<scalar_t, 5>(),                            \
+        output_.packed_accessor32<scalar_t, 5>(),                           \
+        weight_.packed_accessor32<scalar_t, 5>(),                           \
+        bias_ptr,                                                           \
+        stride[0], stride[1], stride[2],                                    \
+        padding[0], padding[1], padding[2],                                 \
+        dilation[0], dilation[1], dilation[2]);                             \
+  }
+
+Tensor conv_depthwise3d_cuda(
+    const Tensor& input,
+    const Tensor& weight,
+    IntArrayRef kernel_size,
+    const Tensor& bias,
+    IntArrayRef stride,
+    IntArrayRef padding,
+    IntArrayRef dilation) {
+  TORCH_CHECK(input.device() == weight.device(), "expects input and weight tensors to be on the same device.");
+  if (bias.defined()) {
+    TORCH_CHECK(input.device() == bias.device(), "expects input and bias tensors to be on the same device.");
+  }
+
+  conv_depthwise_shape_check<3>(input, weight, bias, Tensor() /* undefined */,
+                                kernel_size, stride, padding, dilation);
+
+  auto output_size = conv_output_size(input.sizes(), weight.sizes(),
```

Comment:
conv_output_size assumes that 0-the dimension of input is batch dimension, yet you check input.dim() == 4 only later. It looks like 4d input for 3d convolution is no longer supported - the error is thrown in this case.